### PR TITLE
[Element 2.1] Handle duplicate and expose some browser launch options

### PR DESCRIPTION
**Pull request by honglax**
Relate to this issue https://tricentisflood.atlassian.net/browse/FLO-2045.

### CODE REVIEW

File: packages/core/src/Element.ts [hunk 1]
```diff
@@ -20,19 +31,31 @@ export async function runSingleTestScript(opts: ElementOptions): Promise<Iterati
...
+		downloadsPath,
+	}
+
+	if (!executablePath) {
+		delete launchOptionOverrides.executablePath
+	}
+
+	if (!downloadsPath) {
+		delete launchOptionOverrides.downloadsPath
+	}
```
Comment (phileba) [inline packages/core/src/Element.ts:50]:
```suggestion
	const launchOptionOverrides = {
		headless,
		devtools,
		sandbox,
		browser: browserTypes.includes(browser) ? browser : 'chromium',
		debug: verbose,
	}

	if (executablePath) {
		launchOptionOverrides.executablePath = executablePath
	}

	if (downloadsPath) {
		launchOptionOverrides.downloadsPath = downloadsPath
	}
```

Comment (honglax) [reply packages/core/src/Element.ts:50]: Screen Shot 2020-12-23 at 17 06 25
I've tried and got these errors above

Comment (honglax) [reply packages/core/src/Element.ts:50]: Therefore, I used

```
const launchOptionOverrides = {
		headless,
		devtools,
		sandbox,
		browser: browserTypes.includes(browser) ? browser : 'chromium',
		debug: verbose,
		executablePath,
		downloadsPath,
	}

	if (!executablePath) {
		delete launchOptionOverrides.executablePath
	}

	if (!downloadsPath) {
		delete launchOptionOverrides.downloadsPath
	}
```

Comment (phileba) [reply packages/core/src/Element.ts:50]: yep, this error occurs cause you don't have the interface for the `launchOptionOverrides`, it should be `Partial<ConcreteLaunchOptions>`. Your solution is okay but if the `executablePath` or `downloadsPath` does not exist, we create an object with an undefined value then validate and remove it., I think that's redundant.

Comment (honglax) [reply packages/core/src/Element.ts:50]: Your solution works. Thanks for your suggestion.

Now the code changes to

```
const launchOptionOverrides: Partial<ConcreteLaunchOptions> = {
		headless,
		devtools,
		sandbox,
		browser: browserTypes.includes(browser) ? browser : 'chromium',
		debug: verbose,
	}

	if (executablePath) {
		launchOptionOverrides.executablePath = executablePath
	}
	if (downloadsPath) {
		launchOptionOverrides.downloadsPath = downloadsPath
	}
```

**Review by phileba — APPROVED** [review]

**Review by HieuLeKMS — APPROVED** [review]

File: packages/core/src/utils/BrowserTypes.ts [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
 export const browserTypes: string[] = ['chromium', 'firefox', 'webkit']
 
 export const isCorrectBrowserType = (browser: string): browser is BrowserType =>
 	browserTypes.includes(browser)
+
+export const checkBrowserType = (browser: string): void => {
+	if (!isCorrectBrowserType(browser)) {
+		console.warn(
+			chalk.yellow.bold(
+				`The given browser type '${browser}' must be one in 'chromium', 'firefox', and 'webkit'.\n  Running test scripts with default browser type is 'chromium'.\n`,
```
Comment (HieuLeKMS) [inline packages/core/src/utils/BrowserTypes.ts:13]: `Invalid browser type '${browser}'. Browser type must be 'chromium', 'firefox', or 'webkit', default to 'chromium'.\n`

Comment (honglax) [reply packages/core/src/utils/BrowserTypes.ts]: I've updated the warning message base on the AC4 of [FLO-2045](https://tricentisflood.atlassian.net/browse/FLO-2045)

**Pull request merged**